index.js

```diff
@@ -28,7 +27,52 @@ mongoose.connect(process.env.MONGO_URI, {
     console.error('MongoDB connection error:', err);
 });
 
-const PORT = process.env.PORT || 3000;
+// Multer configuration for file storage
+const storage = multer.diskStorage({
+    destination: (req, file, cb) => {
+        const uploadPath = 'uploads/';
+        if (!fs.existsSync(uploadPath)) {
+            fs.mkdirSync(uploadPath, { recursive: true });
+        }
+        cb(null, uploadPath);
+    },
+    filename: (req, file, cb) => {
+        const fileName = `${file.fieldname}-${Date.now()}${path.extname(file.originalname)}`;
+        cb(null, fileName);
+    }
+});
+
+// File filter for file types
+const fileFilter = (req, file, cb) => {
+    const fileTypes = /jpeg|jpg|png|gif|pdf|doc|docx|csv/;
+    const extname = fileTypes.test(path.extname(file.originalname).toLowerCase());
+    const mimetype = fileTypes.test(file.mimetype);
+    if (mimetype && extname) {
+        return cb(null, true);
+    } else {
+        cb('Error: Allowed file types are jpeg, jpg, png, gif, pdf, doc, docx, csv!');
+    }
+};
+
+// Initialize multer
+const upload = multer({
+    storage: storage,
+    limits: { fileSize: 5 * 1024 * 1024 }, // 5 MB limit
+    fileFilter: fileFilter
+});
+
+// API routes
+const tenderRoutes = require('./routes/tenderRoutes');
+const vendorRouter = require('./routes/vendorRoutes');
+const productRoutes = require('./routes/productRoutes');
+
+// Use API routes
+app.use('/tenders', tenderRoutes);
+app.use('/vendor', vendorRouter);
+app.use('/api/products', productRoutes);
+
+// Start the server
+const PORT = process.env.PORT || 4000;
```

Comment:
WHenever we get network is in use error, we have to check that if the server is getting used in other vs code , or check command prompt to check this